src/config/app-config-endpoints.html

```diff
@@ -1,11 +1,51 @@
 <script>
 
   AppEndpoints = {
-    protocol: 'https://',
-    subdomain: 'subdomain',
-    domain: '.domain.com'
+    apiBase: {
+      protocol: 'https://',
+      subdomain: 'subdomain',
+      apiSuffix: 'api',
+      domain: 'domain.com'
+    }
   };
 
+  function resolveApiBaseUrl() {
+    const config = AppEndpoints.apiBase;
+    let { protocol, port } = window.location;
+    let [ subdomain, ...domain] = window.location.hostname.split('.');
+
+    // Handle vanity domains (no subdomain)
+    if (!domain.includes('.')) {
```

Comment:
There's also these cases:
localhost => 1 part
api.localhost => 2 parts
app.com => 2 parts
api.app.com => 3 parts

So you need something like:
```
const config = AppEndpoints.apiBase;
let { protocol, port='' } = window.location;
let domainParts= window.location.hostname.split('.');

// Find the cases where we have a subdomain, and ensure that they have apiSuffix on the end
if ((domainParts.includes('localhost') && domainParts.length > 1) || domainParts.length > 2) {
  let subdomain = domainParts.shift();
  if (!subdomain.endsWith(config.apiSuffix)) subdomain.concat(`-${config.apiSuffix}`);
  domainParts.unshift(subdomain);
}

// Check to see if we're using localhost
const index = domainParts.findIndex('localhost');
if (index > -1) {
  // We are using localhost, so replace the domain, drop the port, and update the protocol
  domainParts[index] = config.domain;
  protocol = config.protocol;
  port = '';

  // If localhost is the first entry then we want to add our configured subdomain
  if (index === 0) domainParts.unshift(config.subdomain);
}

// Combine our array of domain parts into a string we can use then save it for later
const domain = domainParts.join('.');
config.apiBaseUrl = `${protocol}${domain}${port}`;
```

Note the update to ensure we have a default value set for port, when you run a string literal with undefined as one of the variables it actually drops "undefined" in as part of the string.  Also including subdomain as part of the domainParts ensures '.' is added correctly.